### PR TITLE
updated backend

### PR DESCRIPTION
**Pull request by Iyanuoluwa-abolade**
- Implemented robust server endpoints for user management within the application. These endpoints handle various user-related operations, ensuring seamless interaction between the frontend and backend.

- Implemented session management that enables secure handling and maintaining of a user's session throughout their interaction the app.

Image 7-2-24 at 10 57 AM

Image 7-2-24 at 10 57 AM

### CODE REVIEW

File: backend/index.js [hunk 1]
```diff
@@ -1,10 +1,86 @@
-const { PrismaClient } = require('@prisma/client')
-const prisma = new PrismaClient()
-const cors = require('cors')
-const express = require('express')
+// const { PrismaClient } = require('@prisma/client')
```
Comment (robintan98) [inline backend/index.js]: As we discussed in chat, let's get rid of all the commented code before we merge to main :)

File: backend/index.js [hunk 2]
```diff
@@ -1,10 +1,86 @@
...
+app.use(
+    session({
+        secret: 'SECRET',
+        resave: false,
+        saveUninitialized: false,
+        store: sessionStore,
+        cookie: {
+            sameSite: 'false',
+            secure: false,
+            expires: new Date(Date.now() + 365 * 24 * 60 * 60 * 1000),
```
Comment (robintan98) [inline backend/index.js]: Let's delegate this calculation to another function or re-usable constant to improve readability and cleanliness

File: backend/index.js [hunk 3]
```diff
@@ -1,10 +1,55 @@
...
+    })
+);
+app.use(
+    session({
+        secret: 'SECRET',
+        resave: false,
+        saveUninitialized: false,
+        store: sessionStore,
+        cookie: {
+            sameSite: 'false',
```
Comment (markhopson) [inline backend/index.js:39]: Is it supposed to be a string?

https://expressjs.com/en/resources/middleware/session.html

File: backend/routes/users.js [hunk 4]
```diff
@@ -1,60 +1,77 @@
...
-
-router.post("/users", async (req, res) => {
-    const { username, password } = req.body;
 
-    try {
-        const existingUser = await user.findOne({
-            where: {
-                [Op.or]: [{ username }]
+const router = express.Router();
+const port = 3000;
```
Comment (markhopson) [inline backend/routes/users.js]: Nit: remove if unused.

File: backend/routes/users.js [hunk 5]
```diff
@@ -1,60 +1,77 @@
...
+        });
         if (!user) {
-            return res.status(401).json({ error: "Invalid username or password" });
+            return res.status(401).json({ error: "Invalid credentials" });
+        } else {
+            const isMatch = await bcrypt.compare(Password, user.Password);
+            if (!isMatch) {
+                return res.status(401).json({ error: "Invalid credentials" });
+            }
+            req.session.user = user;
```
Comment (markhopson) [inline backend/routes/users.js:62]: Is this necessary?

File: frontend/my-adopthive-app/src/App.jsx [hunk 6]
```diff
@@ -68,14 +31,11 @@ function App() {
             <Route path="/" element={ user ? < Home /> : <SignIn />} />
             <Route path="/signin" element={<SignIn />} />
             <Route path="signup" element={<SignUp />} />
```
Comment (markhopson) [inline frontend/my-adopthive-app/src/App.jsx]: Leading slash

**Review by markhopson — COMMENTED** [review]
Some comments but otherwise LG

File: backend/index.js (hunk 3 above)
Comment (markhopson) [reply backend/index.js:39]: image

**Review by markhopson — APPROVED** [review]

File: backend/index.js (hunk 3 above)
Comment (Iyanuoluwa-abolade) [reply backend/index.js:39]: The cookie.sameSite is a string that accepts a boolean

File: backend/routes/users.js (hunk 5 above)
Comment (Iyanuoluwa-abolade) [reply backend/routes/users.js:62]: Yes, for maintaining user authentication state

**Pull request merged**